File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(!root) return 0;
        // return max(maxDepth(root -> left), maxDepth(root -> right)) + 1;
        // queue<TreeNode*> q;
        // q.push(root);
        // int h = 0;
        // while(q.size()) {
        //     int sz = q.size();
        //     h++;
        //     for(int i=0; i< sz; i++) {
        //         TreeNode* cur = q.front();
        //         q.pop();
        //         if(cur -> left) q.push(cur -> left);
        //         if(cur -> right) q.push(cur -> right);
        //     }
        // }
        // return h;
        int h = 0;
        stack<pair<TreeNode*, int>> st;
        st.push({root, 1});
        while(st.size()) {
            auto [node, depth] = st.top();
            st.pop();
            h = max(h, depth);
            if(node -> left) st.push({node -> left, depth+1});
            if(node -> right) st.push({node -> right, depth+1});
        }
        return h;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        dq = deque([])
        order = []
        dq.append(root)
        while len(dq) > 0:
            curLen = len(dq)
            tmpList = []
            for i in range(curLen):
                front = dq[0]
                dq.popleft()
                tmpList.append(front.val)
                if front.left:
                    dq.append(front.left)
                if front.right:
                    dq.append(front.right)
            order.append(tmpList)
        return order

